flame/trafgen.cpp

```diff
@@ -42,27 +51,31 @@ TrafGen::TrafGen(std::shared_ptr<uvw::Loop> l,
 void TrafGen::process_wire(const char data[], size_t len)
 {
 
-    ldns_pkt *query{0};
-    int r = ldns_wire2pkt(&query, (uint8_t *)data, len);
-    if (r != LDNS_STATUS_OK) {
+    if (len <= 12) {
         _metrics->bad_receive(_in_flight.size());
-        ldns_pkt_free(query);
         return;
     }
 
-    uint16_t id = ldns_pkt_id(query);
+    struct dns_header *dns_header = (struct dns_header *)(data);
```

Comment:
A couple of observations here.

First, we're casting away the `const` of the passed variable here and then modifying the data below. This is probably a bad idea unless there are very important reasons.

Second, it seems like we only use the query ID and the response code, so we really don't need most of this at all.

I'd suggest something like:
```c
    // The first 2 bytes are the query ID.
    u_int16_t id = ntohs(*(u_int16_t *)data);
    // The response code is in the low-order 4 bits of the flags header field.
    uint8_t rcode = data[3] & 0xf;
```

In general though, it will be nice to remove a dependency and speed up the code at the same time!

Reply:
love it.